File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const numsMap = new Map();
    for (let i = 0; i < nums.length; i++) {
        const value = target - nums[i];
        if (numsMap.has(value)) return [numsMap.get(value), i];
        numsMap.set(nums[i], i);
    }
    return false;
};

File: 1-two-sum/two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    nums_hash = {}

    nums.each_with_index do |num, i|
        value = target-num
        return [nums_hash[value], i]  if(nums_hash[value])
        nums_hash[num] = i
    end
end